### DISCUSSION TITLE
SED configuration

### DISCUSSION THREAD

**Discussion by svenschwermer** [openthread/openthread#8999] [Q&A] [answered]
Hi,

I'm in the process of assessing whether Thread is a suitable protocol for my application. The application is functionally equivalent to a garage opener:
- The receiver side is permanently powered => FTD / border router
- The hand piece ("key fob") would be a battery-powered device that goes out of range frequently and for extended periods of time

For the hand piece I'm thinking to use a SED-type configuration. However, how do I prevent using excessive power when it's out of range? Would it be an option to set the child timeout to a very large value (multiple days) and essentially disable polling? If so, is there an API for disabling polling entirely? That way, the parent wouldn't throw out the child until it eventually comes back and sends payload (button is pressed).

This leaves one challenge: If the hand piece is out of range and the button is pressed, the device will probably lose its network link if doesn't get a response to its message? Can this be prevented somehow?

Thanks,
Sven

**Comment by jwhui** [comment, marked as answer]
A garage door opener typically does not need to take any action until the button is pressed. With that in mind, one option is simply to have the garage door opener attach to the network when the button is pressed, send and receive whatever packets it needs, and then go back to sleep. This would be the most straightforward approach, but you'll have to evaluate whether this fits within your power/latency constraints.

Normally, a child that fails to receive ACKs from its parents will look for a new parent to communicate with. This is part of how Thread implements its self healing mesh topology. The challenge with preventing such behavior is that the child will be restricted to only communicate via a single parent. If this constraint is acceptable, it could be possible to increase the child timeout to a large value and disable the maximum retransmission threshold on the child before it looks for a new parent. This would avoid the cost of having to search and reconnect to a parent on each button press. However, this would not be a typical use of Thread.
